rust/scx_utils/src/topology.rs

```diff
@@ -196,16 +196,27 @@ impl Topology {
             let mut node_cores = BTreeMap::new();
             let mut node_cpus = BTreeMap::new();
 
-            for (&llc_id, llc) in node.llcs.iter_mut() {
+            for (&(mut llc_id), llc) in node.llcs.iter_mut() {
                 let llc_mut = Arc::get_mut(llc).unwrap();
                 let mut llc_cpus = BTreeMap::new();
 
+                // The same LLC ID should not span across multiple NUMA nodes. If it does, create a
+                // new artificial LLC for each node. While this should never happen on physical
+                // hardware, it can occur in virtualized environments.
+                if topo_llcs.contains_key(&llc_id) {
+                    llc_id += 1;
```

Comment:
Would it be easier to understand if this is done in a separate loop before this one? Also, it probably should scan forward and allocate the first unallocated ID to avoid shifting all subsequent IDs and generate a warning message.